crates/bindings-sys/src/lib.rs

```diff
@@ -588,6 +588,15 @@ pub mod raw {
         ///
         /// - `out_ptr` is NULL or `out` is not in bounds of WASM memory.
         pub fn identity(out_ptr: *mut u8);
+
+        /// Check the size of the jwt associated with the given connection.
+        /// Returns 0 if there is no jwt for the connection.
+        pub fn jwt_len(connection_id_ptr: *const u8, out_ptr: *mut u32);
+
+        /// Write the jwt payload for the given connection id to the out_ptr.
+        /// target_ptr_len will be set to the number of bytes written to the buffer.
+        /// If the buffer is too small, or the connection has no jwt, target_ptr_len will be set to 0.
+        pub fn get_jwt(connection_id_ptr: *const u8, target_ptr: *mut u8, target_ptr_len: *mut u32);
```

Comment:
I am surprised that we don't have `get_jwt` return a `BytesSource` which we then call `bytes_source_read` from. I would prefer not to proliferate `read`-like host functions. This would require amending `bytes_source_read` to support `BytesSource`s other than `CALL_REDUCER_ARGS_SOURCE`, and probably also adding `bytes_source_len`, but that seems like valuable work that will need to get done at some point anyways. Notably I will need to do this when I implement HTTP requests in procedures in order to return the response. Because I'm going to be doing it anyways, I could write a patch tomorrow which makes the necessary extensions to `bytes_source_read`.